clients/php/src/ZawgyiUnicodeMarkovModel.php

```diff
@@ -164,15 +164,13 @@ public function predict(string $input, bool $verbose = false): float
         $totalDelta     = 0.0;
         $seenTransition = false;
 
-        for ($offset = 0; $offset <= mb_strlen($input);) {
-            if ($offset == mb_strlen($input)) {
-                $cp        = 0;
-                $currState = 0;
-            } else {
-                $character = mb_substr($input, $offset, 1);
-                $cp = self::ord_utf8($character);
-                $currState = self::getIndexForCodePoint($cp);
-            }
+        $inputBreakIterator = \IntlBreakIterator::createCodePointInstance();
```

Comment:
This makes the code depend on php-intl, which users may not have installed.  UTF-8 decoding should be easy enough to do without depending on php-intl.  Have you thought about a solution that doesn't depend on IntlBreakIterator?

Reply:
Yes, indeed I had a few iterations, including one, with own implementation of UTF-8 encoded string iterator. It had a significant perf improvement over original code too (don't have numbers at hand, but I think it was 1.5x gain), but falling short from one offered by `IntlBreakIterator`. Hence I opted for the most performant and (arguably) simpler code. 
Since I don't know what kind of environment this library is typically used in, I'll let you make a call if introducing a dependency is a right move.

Reply:
OK, I think what I'll do is merge this PR but then bump the major version number of the PHP package so as not to break anyone, and people who don't want the intl dependency can use the previous major version number.